types/react/test/index.ts

```diff
@@ -139,190 +138,186 @@ interface SCProps {
     return props.foo ? React.DOM.div(null, props.foo) : null;
-};
+}
+
+// tslint:disable-next-line:no-namespace
 namespace StatelessComponent {
-    export var displayName = "StatelessComponent";
-    export var defaultProps = { foo: 42 };
+    export const displayName = "StatelessComponent";
+    export const defaultProps = { foo: 42 };
 }
 
-var StatelessComponent2: React.SFC<SCProps> =
+const StatelessComponent2: React.SFC<SCProps> =
     // props is contextually typed
     props => React.DOM.div(null, props.foo);
 StatelessComponent2.displayName = "StatelessComponent2";
 StatelessComponent2.defaultProps = {
     foo: 42
 };
 
-var StatelessComponent3: React.SFC<SCProps> =
+const StatelessComponent3: React.SFC<SCProps> =
     // allows usage of props.children
     // allows null return
     props => props.foo ? React.DOM.div(null, props.foo, props.children) : null;
 
 // React.createFactory
-var factory: React.CFactory<Props, ModernComponent> =
+const factory: React.CFactory<Props, ModernComponent> =
     React.createFactory(ModernComponent);
-var factoryElement: React.CElement<Props, ModernComponent> =
+const factoryElement: React.CElement<Props, ModernComponent> =
     factory(props);
 
-var statelessFactory: React.SFCFactory<SCProps> =
+const statelessFactory: React.SFCFactory<SCProps> =
     React.createFactory(StatelessComponent);
-var statelessFactoryElement: React.SFCElement<SCProps> =
+const statelessFactoryElement: React.SFCElement<SCProps> =
     statelessFactory(props);
 
-var classicFactory: React.ClassicFactory<Props> =
+const classicFactory: React.ClassicFactory<Props> =
     React.createFactory(ClassicComponent);
-var classicFactoryElement: React.ClassicElement<Props> =
+const classicFactoryElement: React.ClassicElement<Props> =
     classicFactory(props);
 
-var domFactory: React.DOMFactory<React.DOMAttributes<{}>, Element> =
-    React.createFactory("foo");
-var domFactoryElement: React.DOMElement<React.DOMAttributes<{}>, Element> =
+const domFactory: React.DOMFactory<React.DOMAttributes<{}>, Element> =
+    React.createFactory("div");
+const domFactoryElement: React.DOMElement<React.DOMAttributes<{}>, Element> =
     domFactory();
 
 // React.createElement
-var element: React.CElement<Props, ModernComponent> =
-    React.createElement(ModernComponent, props);
-var elementNoState: React.CElement<Props, ModernComponentNoState> =
-    React.createElement(ModernComponentNoState, props);
-var statelessElement: React.SFCElement<SCProps> =
-    React.createElement(StatelessComponent, props);
-var classicElement: React.ClassicElement<Props> =
-    React.createElement(ClassicComponent, props);
-var domElement: React.ReactHTMLElement<HTMLDivElement> =
-    React.createElement("div");
-var literalHtmlElement = React.createElement("input", { type: "text" });
-var literalSvgElement = React.createElement("svg", { accentHeight: 12 });
-
-declare let cardhtml: React.HTMLProps<HTMLElement>;
-declare let accessKey: string;
```

Comment:
Readed the test with a correct definition: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/17553/commits/929b4a206daefb5463be587e74ea5fa225c40e90#diff-96b72df8b13a8a590e4f160cbc51f40cR163